eng/packages/http-client-csharp-mgmt/emitter/src/resource-metadata.ts

```diff
@@ -43,6 +43,7 @@ export interface ResourceMetadata {
   resourceScope: ResourceScope;
   parentResourceId?: string;
   singletonResourceName?: string;
+  resourceName?: string;
```

Comment:
I think the `resourceName` should always be there.
Can we change this property to required?
Also in C# code part, the `ResourceName` property should not be nullable.